### DISCUSSION TITLE
Weekly summary of Freenet dev meeting - 15th May, 2023

### DISCUSSION THREAD

**Discussion by sanity** [freenet/freenet-core#629] [General]
# Freenet Development Update

During our weekly meeting, we touched on a range of topics related to the ongoing development of Freenet. Here's a quick summary:

## 1. Decentralized Messaging Application

We've made substantial progress on our first non-trivial decentralized application, a proof-of-concept messaging tool. The toughest part - rendering changes to the inbox - has been tackled. We're currently addressing minor issues related to updating contracts and demonstrated the current state of the application during the meeting.

## 2. Simple Address Book Functionality

We discussed the implementation of a rudimentary address book. At present, user addresses are hard-coded into the system. The introduction of an address book will resolve this, allowing for a more user-friendly experience.

## 3. Message Handling

Discussed a minor UI issue concerning message handling, should be a quick fix.

## 4. Delegates Handling

Discussed delegate's role in storing private data. Once messages are read, they'll be stored in the user's delegate and removed from the inbox contract.

## 5. Contract Summarization and Subscription Model

The contract summarization mechanism and the subscription model issues we addressed are essentially two sides of the same coin. They are integral to how Freenet efficiently synchronizes contract states across multiple computers. If Alice and Bob both have a state of a particular contract, Alice can create a summary of her contract state, a small piece of data she can send to Bob. Bob can then use this summary to create a delta, which will contain the differences between his contract state and Alice's. This delta can then be sent back to Alice very efficiently. The challenge we faced was figuring out how to handle a scenario where Alice doesn't yet have any state. We agreed that if no summary is provided, Bob will send his entire state to Alice.

## 6. Website Update

We've been focusing on updating the Freenet website. Our aim is to enable users to generate a public-private key pair within their browser for enhanced privacy and security. Additionally, we're integrating a donation system that authorizes the public key via a blind signature.

## 7. Next Steps

Our immediate plan is to polish the messaging application to the point where its basic functionality is working and demonstrable. This includes merging changes to the main branch and documenting our progress.

**Comment by sbillig** [comment]
> We've made substantial progress on our first non-trivial decentralized application, a proof-of-concept messaging tool.

Is the repo for this public?

**Reply by sanity** [reply]
@iduartgomez is working on it in a branch: https://github.com/iduartgomez/locutus/tree/inbox-app/apps/freenet-email-app

**Reply by sbillig** [reply]
Thanks!
